src/app/dev/DevToys.Tools/Tools/EncodersDecoders/Base64Text/Base64TextEncoderDecoderCommandLineTool.cs

```diff
@@ -0,0 +1,64 @@
+﻿using DevToys.Tools.Helpers;
+using Microsoft.Extensions.Logging;
+
+namespace DevToys.Tools.Tools.EncodersDecoders.Base64Text;
+
+[Export(typeof(ICommandLineTool))]
+[Name("Base64TextEncoderDecoder")]
+[CommandName(
+    Name = "base64",
+    Alias = "b64",
+    ResourceManagerBaseName = "DevToys.Tools.Tools.EncodersDecoders.Base64Text.Base64TextEncoderDecoder",
+    DescriptionResourceName = nameof(Base64TextEncoderDecoder.Description))]
+internal sealed class Base64TextEncoderDecoderCommandLineTool : ICommandLineTool
+{
+    private enum Conversion
+    {
+        Encode,
+        Decode
+    }
+
+    [CommandLineOption(
+        Name = "input",
+        Alias = "i",
+        IsRequired = true,
+        DescriptionResourceName = nameof(Base64TextEncoderDecoder.InputOptionDescription))]
+    private string? Input { get; set; } // NOTE: Could be a FileInfo instead of string too, if you we want to accept file as input instead.
+
+    [CommandLineOption(
+        Name = "conversion",
+        Alias = "c",
+        IsRequired = true,
+        DescriptionResourceName = nameof(Base64TextEncoderDecoder.ConversionOptionDescription))]
+    private Conversion ConversionMode { get; set; } = Conversion.Encode;
+
+    [CommandLineOption(
+        Name = "encoding",
+        Alias = "e",
+        DescriptionResourceName = nameof(Base64TextEncoderDecoder.EncodingOptionDescription))]
+    private Base64Encoding EncodingMode { get; set; } = Base64Encoding.Utf8;
+
+    public ValueTask<int> InvokeAsync(ILogger logger, CancellationToken cancellationToken)
+    {
+        string output;
+        if (ConversionMode == Conversion.Encode)
```

Comment:
I think we should share this block of code between the Command Line and the GUI tool no?

Reply:
Good point!

Reply:
Actually, nevermind. In the GUI tool, the parameter to figure out whether we Encode or Decode is a boolean. It's practical this way because the Switch() UI Element accepts a boolean for determining the state of the toggle switch.
Meanwhile, the enum is a better fit for Command Line because otherwise the user would have to type "true" or "false" when writing the parameters when invoking the DevToys.CLI app.

Let's keep it as it is then,